src/index.js

```diff
@@ -122,20 +143,66 @@ class EdfContentScript extends ContentScript {
 
   async fetch(context) {
     this.log('info', 'fetch start')
+    const { sourceAccountIdentifier, manifest, trigger } = context
+
+    // force fetch all data (the long way) when last trigger execution is older than 30 days
+    // or when the last job was an error
+    const isLastJobError =
+      trigger.current_state.last_failure > trigger.current_state.last_success
+    const distanceInDays = getDateDistanceInDays(
+      trigger.current_state.last_execution
+    )
+    if (distanceInDays >= 30 || isLastJobError) {
+      this.log('debug', `isLastJobError: ${isLastJobError}`)
+      this.log('debug', `distanceInDays: ${distanceInDays}`)
+      FORCE_FETCH_ALL = true
+    }
+
     if (this.store && this.store.email && this.store.password) {
+      this.log('info', 'saving credentials')
       await this.saveCredentials(this.store)
     }
     const contact = await this.fetchContact()
     const contracts = await this.fetchContracts()
     await this.fetchAttestations(contracts, context)
     await this.fetchBillsForAllContracts(contracts, context)
     const echeancierResult = await this.fetchEcheancierBills(contracts, context)
-    const housing = formatHousing(
-      contracts,
-      echeancierResult,
-      await this.fetchHousing()
-    )
-    await this.saveIdentity({ contact, housing })
+
+    // fetch the housing data only if we do not have an existing identity or if the existing
+    // identity is older than 1 month
+    let lastIdentityUpdatedSinceDays = Infinity
+    if (!FORCE_FETCH_ALL) {
+      const existingIdentities = await this.queryAll(
```

Comment:
Same question here, will it not be better to do the query directly in the lib ?
That way no need to import "Q" in each konnectors, we'll juste need to call a `queryAllIdentities` or `queryAllFiles` (with or without arguments, depending on how we can achieve this) and just get an Array back to check ?

Reply:
There are pros and cons :

pros : 
 - what you just said

cons:
	- this gives a lot less flexibility, if the konnector just needs the last file, we would still need to query all the files and this is bad for performance of the konnector
 - this would be to different from the cozy-client api we want to promote and for which the konnectors would be a good "easy" try

That is why we chose to be as much as possible close to the cozy-client api.